static/app/utils/discover/eventView.tsx

```diff
@@ -431,6 +431,18 @@ class EventView {
     return EventView.fromSavedQuery(saved);
   }
 
+  static fromNewQueryWithPageFilters(newQuery: NewQuery, pageFilters: PageFilters) {
+    return EventView.fromSavedQuery({
+      ...newQuery,
+      environment: newQuery.environment ?? pageFilters.environments,
+      projects: newQuery.projects ?? pageFilters.projects,
+      start: newQuery.start ?? pageFilters.datetime.start ?? undefined,
+      end: newQuery.end ?? pageFilters.datetime.end ?? undefined,
+      range: newQuery.range ?? pageFilters.datetime.period ?? undefined,
+      utc: newQuery.utc ?? pageFilters.datetime.utc ?? undefined,
+    });
```

Comment:
I'm not too sure about this, you might have to explain some of this to me later. But I'm not sure why we can't reliably get this from the `location`. Or are you proposing to deprecate the `fromQueryWithLocation` and use `fromQueryWithPageFilters`? Imo the location should always match the location and vise versa

Reply:
🤔 I put this PR together because I saw two patterns I don't like, and replaced them with one pattern I _do_ like, that's all! I'm not really intending to deprecate anything, or have any other agenda

The first pattern I don't like is passing both `location` and `pageFilters` to functions. Long parameter lists are a code smell. Redundant or overlapping parameters (`location` and `pageFilters` store basically the same information) are a _very_ stinky code smell. So, I thought it'd be good to change the code to only need one of them.

The second pattern I don't like is using `location` to pull the project ID, start/end dates, and environment. `PageFilters` is a smart abstraction over `location`, _purpose-made_ to store the global selection state! It has all kinds of clever tricks like using `localStorage` to store/fetch the state, listening to changes, etc. Why wouldn't we _also_ use it?

So, I between using `location` and `PageFilters`, I chose `PageFilters`. Having done that, I was a little annoyed that there's no way to easily create an `EventView` from them, so I added a new factory way to do it

That's it, really, nothing else major going on!

Reply:
Makes sense to me and thanks for the clarification! 
I definitely agree with your comment about these two patterns. I'm more curious on what the future of `fromQueryWithLocation` is, I know your intent wasn't to dictate in this PR. But imo it would be good to stick to `fromQueryWithPageFilters` for everything in the future!

Reply:
Fair!